src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/AdoNet/AdoNetClientInstrumentMethodAttribute.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet
+{
+    internal class AdoNetClientInstrumentMethodAttribute : InstrumentMethodAttribute
+    {
+        public AdoNetClientInstrumentMethodAttribute(Type adoNetClientDataType)
+        {
+            AdoNetClientData = (IAdoNetClientData)Activator.CreateInstance(adoNetClientDataType);
+
+            Assembly = AdoNetClientData.AssemblyName;
+            Type = AdoNetClientData.SqlCommandType;
+            MinimumVersion = AdoNetClientData.MinimumVersion;
+            MaximumVersion = AdoNetClientData.MaximumVersion;
+            IntegrationName = AdoNetClientData.IntegrationName;
+        }
+
+        protected IAdoNetClientData AdoNetClientData { get; }
+
+        internal class CommandExecuteNonQueryAsyncAttribute : AdoNetClientInstrumentMethodAttribute
+        {
+            public CommandExecuteNonQueryAsyncAttribute(Type adoNetClientDataType)
+                : base(adoNetClientDataType)
+            {
+                Method = AdoNetConstants.MethodNames.ExecuteNonQueryAsync;
+                ReturnTypeName = AdoNetConstants.TypeNames.IntTaskType;
+                ParametersTypesNames = new[] { ClrNames.CancellationToken };
```

Comment:
Not directly related to this PR, but this property should be called `ParameterTypeNames` (only `Names` needs to be plural).

Reply:
Also `MethodName` and `CallTargetTypeName`, for consistency.

Reply:
done in a93295c. CallTargetClass has been changed to CallTargetClassType.

Reply:
Oh yeah, that one is a `Type`, not a `string`. Thanks 😉 
Maybe `CallTargetType` then?

Reply:
😂 ok, I change it, I don't have a strong opinion.

Reply:
thanks!